pkg/api/validation/schema.go

```diff
@@ -88,6 +101,29 @@ func (s *SwaggerSchema) ValidateBytes(data []byte) error {
 	if kind == nil {
 		return fmt.Errorf("kind not set")
 	}
+	if kind.(string) == "List" {
```

Comment:
I would really prefer if we did the check based on the semantics of the provided object (runtime.IsList) - we do that in most other places in the code instead of hardcoding list.